### PR TITLE
[libc] Implement process_mrelease.

### PR DESCRIPTION
**Pull request by moar55**
This PR addresses #110124.

### CODE REVIEW

**Comment by moar55** [comment]
Hey @SchrodingerZhu can you assign a CO(if you aren't already) to view my PR :)

File: libc/src/sys/mman/linux/process_mrelease.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace LIBC_NAMESPACE_DECL {
+
+LLVM_LIBC_FUNCTION(int, process_mrelease, (int pidfd, unsigned int flags)) {
+  long ret =
+      LIBC_NAMESPACE::syscall_impl<int>(SYS_process_mrelease, pidfd, flags);
+
+  if (ret < 0) {
+    libc_errno = static_cast<int>(-ret);
+    return libc_errno;
```
Comment (SchrodingerZhu) [inline libc/src/sys/mman/linux/process_mrelease.cpp:27]: According to https://github.com/bminor/glibc/blob/d92a5e1dad39ab0ac09ab1beeb055c1f1d981cd4/sysdeps/unix/sysv/linux/tst-process_mrelease.c#L50, `-1` is expected on error

File: libc/src/sys/mman/process_mrelease.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
+//===-- Implementation header for process_mrelease  function -----------------*-
```
Comment (SchrodingerZhu) [inline libc/src/sys/mman/process_mrelease.h:1]: Please fix the license header

File: libc/test/src/sys/mman/linux/process_mrelease_test.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
+//===-- Unittests for process_mrelease
```
Comment (SchrodingerZhu) [inline libc/test/src/sys/mman/linux/process_mrelease_test.cpp:1]: ditto

File: libc/test/src/sys/mman/linux/process_mrelease_test.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    ASSERT_EQ(LIBC_NAMESPACE::process_mrelease(pidfd, 0), EINVAL);
+
+    LIBC_NAMESPACE::close(pidfd);
+  }
+}
+
+TEST(LlvmLibcMProcessMReleaseTest, ErrorNonExistingPidfd) {
+
+  ASSERT_EQ(LIBC_NAMESPACE::process_mrelease(12345, 0), EBADF);
```
Comment (SchrodingerZhu) [inline libc/test/src/sys/mman/linux/process_mrelease_test.cpp:71]: What if `12345` exists? Maybe it is better to use `-1` instead.

**Review by SchrodingerZhu — CHANGES_REQUESTED** [review]

File: libc/test/src/sys/mman/linux/process_mrelease_test.cpp (hunk 4 above)
Comment (moar55) [reply libc/test/src/sys/mman/linux/process_mrelease_test.cpp:78]: good point :)

File: libc/test/src/sys/mman/linux/process_mrelease_test.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
+//===-- Unittests for process_mrelease ------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//===-- Unittests for process_mrelease
```
[the remark this answers is not in the document]
Comment (moar55) [reply libc/test/src/sys/mman/linux/process_mrelease_test.cpp]: oops

**Review by SchrodingerZhu — APPROVED** [review]
LGTM. Thank you for the patch!

**Pull request merged**